src/Compilers/Core/CodeAnalysisTest/DefaultAnalyzerAssemblyLoaderTests.cs

```diff
@@ -30,63 +30,131 @@
 
 namespace Microsoft.CodeAnalysis.UnitTests
 {
-    [CollectionDefinition(Name)]
-    public class AssemblyLoadTestFixtureCollection : ICollectionFixture<AssemblyLoadTestFixture>
-    {
-        public const string Name = nameof(AssemblyLoadTestFixtureCollection);
-        private AssemblyLoadTestFixtureCollection() { }
-    }
+#if NETCOREAPP
 
-    public sealed class DefaultAnalyzerAssemblyLoaderTests : TestBase
+    public sealed class InvokeUtil
     {
-#if NETCOREAPP
-        private sealed class InvokeUtil
+        public void Exec(Action<string> testOutputHelper, AssemblyLoadContext alc, bool shadowLoad, string typeName, string methodName)
         {
-            public void Exec(string typeName, string methodName) => InvokeTestCode(typeName, methodName);
+            // Ensure that the test did not load any of the test fixture assemblies into 
+            // the default load context. That should never happen. Assemblies should either 
+            // load into the compiler or directory load context.
+            //
+            // Not only is this bad behavior it also pollutes future test results.
+            var count = AssemblyLoadContext.Default.Assemblies.Count();
+            using var fixture = new AssemblyLoadTestFixture();
+            using var tempRoot = new TempRoot();
+            var loader = shadowLoad
+                ? new ShadowCopyAnalyzerAssemblyLoader(alc, tempRoot.CreateDirectory().Path)
+                : new DefaultAnalyzerAssemblyLoader(alc);
+            try
+            {
+                DefaultAnalyzerAssemblyLoaderTests.InvokeTestCode(loader, fixture, typeName, methodName);
+            }
+            finally
+            {
+                testOutputHelper($"Test fixture root: {fixture.TempDirectory.Path}");
```

Comment:
It wasn't clear to me why we use this action instead of passing the ITestOutputHelper itself and calling WriteLine.

Reply:
It's complicated ... 

The way the `AssemblyLoadContext` are setup today there are two copies of `ITestOutputHelper` because `XUnit.Abstractions.dll` is loaded twice: once in the default ALC and once in the custom ALC. That means if you try and pass `ITestOutputHelper` directly here you get a type mismatch at runtime because they are different types. 

To pass it around I need to move all of the xUnit loading into the default ALC. That is a bit challenging given how it's setup. I'm trying experiments to do that but may push that off into a follow up PR.